### PR TITLE
homework7

### PR DESCRIPTION
**Pull request by Karl5236543**

### CODE REVIEW

File: homework7.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        self._current_dir = os.getcwd()
+
+    def __enter__(self):
+        os.chdir(self._path)
+    
+    def __exit__(self, exc_type, exc_value, tb):
+        os.chdir(self._current_dir)
+        if self._suppress_exc:
+            return exc_type is not None and issubclass(exc_type, self._suppress_exc)
+
```
Comment (katevoskresenska) [inline homework7.py:25]: this task was implemented incorrectly, you should send an error to the context manager.
Example :

with cd('tmp', FileNotFoundError) that FileNotFoundError should be suppressed
or
with cd('tmp.txt', NotADirectoryError) that NotADirectoryError should be suppressed

File: homework7.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    os.chdir(path)
+    try:
+        yield
+    except:
+        exc_type = sys.exc_info()[0]
+        if not (suppress_exc and issubclass(exc_type, suppress_exc)):
+            raise
+    finally:
+        os.chdir(current_dir)
+
```
Comment (katevoskresenska) [inline homework7.py:46]: The same thing here. Such decorator works if the path exists only.

File: homework7.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            raise ValueError("директория не найдена")
+        self._is_suppress = True if suppress_exc else False
+        self._suppress_exc = suppress_exc or Exception
+        self._path = path
+        self._current_dir = os.getcwd()
+
+    def __enter__(self):
+        try:
+            os.chdir(self._path)
+            raise IndexError()
```
Comment (mmykhailenko) [inline homework7.py:18]: why we raise IndexError after change dir ?

File: homework7.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import contextlib
+import sys, os
+import time
+
+
+class cd:
+    def __init__(self, path, suppress_exc=None):
+        if not os.path.exists(path):
+            raise ValueError("директория не найдена")
```
Comment (mmykhailenko) [inline homework7.py:9]: Not sure why we have here ValueError. Should be FileNotFoundError

**Pull request closed without merging**